benchmark/cxx-source/CxxVectorSum.swift

```diff
@@ -0,0 +1,89 @@
+//===--- CxxVectorSum.swift -----------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2012 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+// This is a benchmark that tracks how quickly Swift can sum up a C++ vector
+// as compared to the C++ implementation of such sum.
+
+import TestsUtils
+import CxxStdlibPerformance
+import Cxx
+
+public let benchmarks = [
+  BenchmarkInfo(
+      name: "CxxVectorOfU32SumInCxx",
+      runFunction: run_CxxVectorOfU32SumInCxx,
+      tags: [.validation, .bridging, .cxxInterop],
+      setUpFunction: create_CxxVectorOfU32),
+  BenchmarkInfo(
+    name: "CxxVectorOfU32SumInSwift_Fastest",
+    runFunction: run_CxxVectorOfU32SumInSwift_Fastest,
+    tags: [.validation, .bridging, .cxxInterop],
+    setUpFunction: create_CxxVectorOfU32),
+  BenchmarkInfo(
+    name: "CxxVectorOfU32SumInSwift",
+    runFunction: run_CxxVectorOfU32SumInSwift,
+    tags: [.validation, .bridging, .cxxInterop],
+    setUpFunction: create_CxxVectorOfU32)
+]
+
+// FIXME: compare CxxVectorOfU32SumInCxx to CxxVectorOfU32SumInSwift and
+// establish an expected threshold of performance, which when exceeded should
+// fail the benchmark.
+
+var vectorOfU32: VectorOfU32!
```

Comment:
Maybe make this a non-optional and fill it in the create/setup function?

Reply:
Global vars aren't super common and are very hard to optimize. I guess there's no way to make it a let?

Reply:
Maybe we could make the vector smaller, and just do more iterators:
```
for _ in n * 1000 {
    for i in vectorOfU32 {
```

Reply:
Yes, let's try that.